File: Data Structures & Algorithms/majority-element/submission-1.java
class Solution {
    public int majorityElement(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();

        for (int num: nums){
            map.put(num, map.getOrDefault(num, 0)+1);
        }

        int max_val = 0;
        int max_freq = 0;

        for(Map.Entry<Integer, Integer> e: map.entrySet()){
            int key = e.getKey();
            int freq = e.getValue();
            if(freq > max_freq){
                max_freq = freq;
                max_val = key;
            }
        }
        return max_val;
    }
}

File: Data Structures & Algorithms/majority-element-ii/submission-1.py
class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        nums.sort()
        output, n = [], len(nums)

        i = 0
        while i < n:
            j = i + 1
            while j < n and nums[i] == nums[j]:
                j += 1
            if (j-i) > n // 3:
                output.append(nums[i])
            i = j
        
        return output